tests/unit/integration_test.py

```diff
@@ -0,0 +1,127 @@
+import re
```

Comment:
```console
% rflx check test.rflx        
Parsing test.rflx                                                                                                                                                                                                   
                                                     
------------------------------ RecordFlux Bug ------------------------------                                                                                                                                        
RecordFlux 0.6.0-pre        
RecordFlux-parser 0.9.0                                                                                                                                                                                             
attrs 21.2.0                                     
icontract 2.5.5                                                                                                                                                                                                     
pydotplus 2.0.2                                                                                           
ruamel.yaml 0.17.16                                                                                                                                                                                                 
z3-solver 4.8.12.0                                  
                                                                                                                                                                                                                    
Optimized: True             
                                                                                                                                                                                                                    
Command: /home/devel/Componolit/RecordFlux/venv/bin/rflx check test.rflx
Traceback (most recent call last):                                                                                                                                                                                  
  File "/home/devel/Componolit/RecordFlux/rflx/cli.py", line 198, in main                                                                                                                                           
    args.func(args)                                                                                                                                                                                                 
  File "/home/devel/Componolit/RecordFlux/rflx/cli.py", line 239, in check                                                                                                                                          
    parse(args.files, args.no_verification, args.workers)                                                                                                                                                           
  File "/home/devel/Componolit/RecordFlux/rflx/cli.py", line 284, in parse                                                                                                                                          
    parser.parse(*present_files)                                                                                                                                                                                    
  File "/home/devel/Componolit/RecordFlux/rflx/specification/parser.py", line 1553, in parse                                                                                                                        
    error.extend(self.__parse_specfile(f))                                                                                                                                                                          
  File "/home/devel/Componolit/RecordFlux/rflx/specification/parser.py", line 1519, in __parse_specfile                                                                                                             
    content = yaml.load(integration_file)                                                                                                                                                                           
  File "/home/devel/Componolit/RecordFlux/venv/lib/python3.9/site-packages/ruamel/yaml/main.py", line 431, in load                                                                                                  
    return self.load(fp)                                                                                                                                                                                            
  File "/home/devel/Componolit/RecordFlux/venv/lib/python3.9/site-packages/ruamel/yaml/main.py", line 434, in load                                                                                                  
    return constructor.get_single_data()                                                                                                                                                                            
  File "/home/devel/Componolit/RecordFlux/venv/lib/python3.9/site-packages/ruamel/yaml/constructor.py", line 122, in get_single_data                                                                                
    return self.construct_document(node)                                                                                                                                                                            
  File "/home/devel/Componolit/RecordFlux/venv/lib/python3.9/site-packages/ruamel/yaml/constructor.py", line 132, in construct_document                                                                             
    for _dummy in generator:                                                                                                                                                                                        
  File "/home/devel/Componolit/RecordFlux/venv/lib/python3.9/site-packages/ruamel/yaml/constructor.py", line 1617, in construct_yaml_map                                                                            
    self.construct_mapping(node, data, deep=True)                                                                                                                                                                   
  File "/home/devel/Componolit/RecordFlux/venv/lib/python3.9/site-packages/ruamel/yaml/constructor.py", line 1500, in construct_mapping                                                                             
    value = self.construct_object(value_node, deep=deep)                                                                                                                                                            
  File "/home/devel/Componolit/RecordFlux/venv/lib/python3.9/site-packages/ruamel/yaml/constructor.py", line 155, in construct_object                                                                               
    data = self.construct_non_recursive_object(node)                                                                                                                                                                
  File "/home/devel/Componolit/RecordFlux/venv/lib/python3.9/site-packages/ruamel/yaml/constructor.py", line 197, in construct_non_recursive_object                                                                 
    for _dummy in generator:                                                                                                                                                                                        
  File "/home/devel/Componolit/RecordFlux/venv/lib/python3.9/site-packages/ruamel/yaml/constructor.py", line 1617, in construct_yaml_map                                                                            
    self.construct_mapping(node, data, deep=True)                                                                                                                                                                   
  File "/home/devel/Componolit/RecordFlux/venv/lib/python3.9/site-packages/ruamel/yaml/constructor.py", line 1501, in construct_mapping                                                                             
    if self.check_mapping_key(node, key_node, maptyp, key, value):                                                                                                                                                  
  File "/home/devel/Componolit/RecordFlux/venv/lib/python3.9/site-packages/ruamel/yaml/constructor.py", line 295, in check_mapping_key                                                                              
    raise DuplicateKeyError(*args)                                                                                                                                                                                  
ruamel.yaml.constructor.DuplicateKeyError: while constructing a mapping                                                                                                                                             
  in "test.rfi", line 2, column 5                                                                                                                                                                                   
found duplicate key "Session" with value "ordereddict([('Buffer_Siz', ordereddict([('Default', 2048), ('Global', ordereddict([('Message', 1024)])), ('Local', ordereddict([('Next', ordereddict([('M', 8192)]))]))])
)])" (original value: "ordereddict([('Buffer_Siz', ordereddict([('Default', 2048), ('Global', ordereddict([('Message', 1024)])), ('Local', ordereddict([('Next', ordereddict([('M', 8192)]))]))]))])")
  in "test.rfi", line 10, column 5
```

Reply:
Can you please provide me with the input you used for this? Thanks!

Reply:
> found duplicate key "Session"

Just duplicate any key in the YAML file.

Reply:
Right, we don't protect against any kind of syntax errors (or other errors when loading the file). We need to add a try-catch there that propagates the relevant error into the error object there.

Reply:
Fixed in 08de02fd54a8b67246fe527c391c005ebf7a98bc